### PR TITLE
Added Azure.Arc.Server.MaintenanceConfig

### LINKED ISSUE #2122: Use a VM maintenance configuration for Arc enabled servers (preview)
### Existing rule

_No response_

### Suggested rule

Arc enabled servers can be assigned a maintenance configuration which allows managed assessments and updates for VM patches within the guest operating system.

This can be done by creating a `Microsoft.Maintenance/configurationAssignments` sub resource with the linked maintenance configuration resource Id.

For in-flight resources we need to collect and export this data.

### Pillar

Operational Excellence

### Additional context

- https://learn.microsoft.com/azure/templates/microsoft.maintenance/configurationassignments
- https://learn.microsoft.com/azure/update-center/manage-arc-enabled-servers-programmatically
- https://learn.microsoft.com/azure/update-center/manage-update-settings

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2122 

Added Azure.Arc.Server.MaintenanceConfig rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Lemme know what you're thinking.

File: docs/en/rules/Azure.Arc.Server.MaintenanceConfig.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Use a maintenance configuration for Arc-enabled servers.
+
+## DESCRIPTION
+
+Arc-enabled servers can be attached to a maintenance configuration which allows customer managed assessments and updates for machine patches within the guest operating system.
+
+## RECOMMENDATION
+
+Consider associating a maintenance configuration for Arc-enabled servers.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.Arc.Server.MaintenanceConfig.md:21]:
```suggestion
Consider automatically managing and applying operating system updates with a maintenance configuration.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Nice work @BenjaminEngeset. I've added a few suggestions.

Let me know if you have any questions.

**Comment by BenjaminEngeset** [comment]
@BernieWhite Thanks for the great feedback. I've adjusted accordingly now :)

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**